src/Context.tsx

```diff
@@ -4,6 +4,8 @@ import { ResolveFunction } from "./Get";
 export interface RestfulReactProviderProps<T = any> {
   /** The backend URL where the RESTful resources live. */
   base: string;
+  /** The current path the request will attempt to resolve. */
+  url?: string;
```

Comment:
🤔 I don't understand what it's suppose to do.

Reply:
Previously we were using the base to collect up the nested routes, for example if we had a base of `https://example.com` and a structure like so:

```jsx
<Get path="/first">
  {() => (<Get path="/second">{children}</Get>)}
</Get>
```

The base when rendering the nested `Get` would be `https://example.com/first` which means we lose the context of the original base url. But in order to maintain a nested url structure we need to carry the routes as we render more nested `Get`s. The url is what that is for. So instead of mutating the original base we append to a string of the relative location.

For the example above the base would remain the same `https://example.com` and our url variable will have a value of `/first`. if we decided to render a third `Get` our url will look like `/first/second` and so on.

Does that help? Is my comment in the code what is causing the confusion or is it that the url variable may not be necessary?

Reply:
But we are on `RestfulReactProvider` so if I'm writting this:
```jsx
<RestfulReactProvider base="https://example.com">
<Get path="/plop" /> /* fetch https://example.com/plop */
</RestfulReactProvider>
```

But this:
```jsx
<RestfulReactProvider base="https://example.com" url="/plop" >
<Get path="/plop" /> /* fetch ??? */
</RestfulReactProvider>
```

On my opinion `base` and `url` are the same here, so it's really confusing

Reply:
The goal in my memory was to have this:

```jsx
<RestfulReactProvider base="https://example.com">
   <Get path="/plop" /> /* fetch https://example.com/plop */
   <Get path="/plop">{() => (
      <Get path="/foo" /> /* fetch https://example.com/plop/foo */
      <Get path="foo" /> /* fetch https://example.com/foo */
    )}</Get>
</RestfulReactProvider>
```

So I don't understand why we add a `url` props to the provider.

Reply:
I see what you are saying with the url property. I do agree that you will have unexpected behavior if you manually pass it as a prop.

My reasoning for using it was that we do not know what the original base url is once nested because it gets mutated.

so if I started off with a base url of: `https://example.com/v1/api` then have you structure:

```jsx
<RestfulReactProvider base="https://example.com/v1/api">
   <Get path="/plop" /> /* fetch https://example.com/v1/api/plop */
   <Get path="/plop">{() => (
      /* this one here is the issue, we don't know what the original base was because it gets mutated with the nested path */
      <Get path="/foo" /> /* expected fetch https://example.com/v1/api/foo  reality fetch https://example.com/v1/api/plop/foo */
      <Get path="foo" /> /* expected fetch https://example.com/v1/api/plop/foo */
    )}</Get>
</RestfulReactProvider>
```

Reply:
the `url` prop is really internal only and shouldn't be exposed as part of the API, I can see through your example how it could cause confusion.

Reply:
I don't want to make the assumption about what a base url looks like, ie.

`https://example.com/v1/api` is a valid base.

Is that a good understanding @fabien0102 ?

Reply:
Without assuming the structure of the base url I cannot think of a way of using the props that exist. We need some way of knowing what the expected default is.

We can be opinionated and assume that a base url is exactly: `http(s)://[url-ish]/`

We are equating the route host with the base url prop.

Is that the correct assumption?

Reply:
Since `RestfulReactProvider` and `Get` derived from `React.Context`, you can simply store the `base` and the `nestedBase` inside the context and consume these both variables without assuming anything.

A think that you should study a bit more the actual structure of the app ;) 
https://reactjs.org/docs/context.html

Reply:
Sorry @fabien0102 I may have been worried about adding more props to the interface. The new push should have resolved this. It adds the original base to the context